bin/release-package.php

```diff
@@ -72,8 +85,34 @@
 );
 execute( $command, 'Could not add the new package repository remote.', true, true );
 
+// Update the version numbers of each dependency of the package we are releasing.
+execute( 'bin/version-packages.sh --no-update', 'Could not update sub-package dependency versions.', true, true );
+
+// Create a new branch to prepare our release.
+$release_branch = sprintf(
+	'update/jetpack-%1$s-v%2$s',
+	$package_name,
+	$tag_version
+);
+$command        = sprintf(
+	'git checkout -b %1$s',
+	escapeshellarg( $release_branch )
+);
+execute( $command, 'Could not create new release branch.' );
+
+// Commit those changes.
+$command = sprintf(
+	'git add composer.json && ( git diff-index --quiet HEAD || git commit -m "Updating dependencies for %1$s" )',
```

Comment:
what's the value in `git diff-index --quiet HEAD`?

Reply:
It's from #14201. It allows us to not try to commit if there are no changes staged for commit.